lib/ui/home/home_screen.dart

```diff
@@ -32,6 +68,7 @@ class _HomeScreenState extends ConsumerState<HomeScreen> {
           getUserProfileSuccess: (profile) => {
                 ref.read(_profileAvatarProvider.notifier).state =
                     profile.avatarUrl,
+                ref.read(_loadingStateProvider.notifier).state = false,
```

Comment:
@doannimble The loading state here is used for the Home header to display the avatar as it was already completed on the previous sprint.

Reply:
@hoangnguyen92dn I think it is not the best solution. If we display the home screen with many async parts, we have to define the state for each.
In case they use the same state to display only one loading, we have to sync these parts inside the ViewModel 🤔

Reply:
@doannimble As per comment from @manh-t https://github.com/hoangnguyen92dn/survey-flutter-ic/pull/55#discussion_r1157993058, I refactored the Provider in ViewModel to handle with `StreamController` 😄 Please kindly check again 🙏